plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsClient.java

```diff
@@ -208,26 +249,52 @@ private Credential getCredentials()
         }
     }
 
+    public String getCachedSheetExpressionForTable(String tableName)
+    {
+        Optional<String> sheetExpression = tableSheetMappingCache.getUnchecked(tableName);
+        if (sheetExpression.isEmpty()) {
+            throw new TrinoException(SHEETS_UNKNOWN_TABLE_ERROR, "Sheet expression not found for table " + tableName);
+        }
+        return sheetExpression.get();
+    }
+
     private List<List<Object>> readAllValuesFromSheetExpression(String sheetExpression)
     {
+        SheetsSheetIdAndRange sheetIdAndRange = new SheetsSheetIdAndRange(sheetExpression);
         try {
-            // by default loading up to 10k rows from the first tab of the sheet
-            String defaultRange = "$1:$10000";
-            String[] tableOptions = sheetExpression.split("#");
-            String sheetId = tableOptions[0];
-            if (tableOptions.length > 1) {
-                defaultRange = tableOptions[1];
-            }
-            log.debug("Accessing sheet id [%s] with range [%s]", sheetId, defaultRange);
-            return sheetsService.spreadsheets().values().get(sheetId, defaultRange).execute().getValues();
+            log.debug("Accessing sheet id [%s] with range [%s]", sheetIdAndRange.getSheetId(), sheetIdAndRange.getRange());
+            return sheetsService.spreadsheets().values().get(sheetIdAndRange.getSheetId(), sheetIdAndRange.getRange()).execute().getValues();
         }
         catch (IOException e) {
             throw new TrinoException(SHEETS_UNKNOWN_TABLE_ERROR, "Failed reading data from sheet: " + sheetExpression, e);
         }
     }
 
-    private static CacheBuilder<Object, Object> newCacheBuilder(long expiresAfterWriteMillis, long maximumSize)
+    public Integer insertIntoSheet(String sheetExpression, List<List<Object>> rows)
```

Comment:
Returned value is unsued.

Reply:
Removed the return value. I didn't have enough knowledge of Trino to know, if this information should be passed to somewhere else